File: core/widget/src/commonMain/kotlin/now/shouldigooutside/core/widget/WidgetConfig.kt
package now.shouldigooutside.core.widget

import kotlinx.serialization.Serializable

@Serializable
public data class WidgetConfig(
    val backendUrl: String,
    val clientId: String,
    val lat: Double,
    val lon: Double,
    val locationName: String,
    val minTemp: Int,
    val maxTemp: Int,
    val maxWind: Int,
    val allowRain: Boolean,
    val allowSnow: Boolean,
    val maxAqi: Int,
    val includeAirQuality: Boolean,
    val activityName: String = WidgetData.DEFAULT_ACTIVITY_NAME,
)

File: apps/ios/SIGOWidget/WidgetConfig.swift
import Foundation

struct WidgetConfig: Codable {
    let backendUrl: String
    let clientId: String
    let lat: Double
    let lon: Double
    let locationName: String
    let minTemp: Int
    let maxTemp: Int
    let maxWind: Int
    let allowRain: Bool
    let allowSnow: Bool
    let maxAqi: Int
    let includeAirQuality: Bool
    let activityName: String?

    static func load() -> WidgetConfig? {
        guard let defaults = UserDefaults(suiteName: "group.now.shouldigooutside"),
              let jsonString = defaults.string(forKey: "widget_config_json"),
              let data = jsonString.data(using: .utf8) else {
            return nil
        }
        return try? JSONDecoder().decode(WidgetConfig.self, from: data)
    }
}
